torch/csrc/jit/type.cpp

```diff
@@ -248,10 +255,20 @@ TypePtr matchTypeVariables(TypePtr formal, TypePtr actual, TypeEnv& type_env) {
   } else if (auto opt_formal = formal->cast<OptionalType>()) {
     if (auto opt_actual = actual->cast<OptionalType>()) {
       return OptionalType::create(matchTypeVariables(
-          opt_formal->getElementType(), opt_actual->getElementType(), type_env));
+          opt_formal->getElementType(),
+          opt_actual->getElementType(),
+          type_env));
     } else {
+      if (opt_formal->getElementType()->kind() == TypeKind::VarType) {
+        return OptionalType::create(
+            matchTypeVariables(opt_formal->getElementType(), actual, type_env));
+      }
+      if (actual->isSubtypeOf(NoneType::get())) {
```

Comment:
I don't think we can reach to this `if` in any cases, can you provide me a example?